projects/packages/assets/src/class-script-data.php

```diff
@@ -86,17 +87,18 @@ public static function render_script_data() {
 		self::$did_render_script_data = true;
 
 		$script_data = is_admin() ? self::get_admin_script_data() : self::get_public_script_data();
-
-		$script_data = wp_json_encode(
+		$script_json = wp_json_encode(
 			$script_data,
 			JSON_UNESCAPED_SLASHES | JSON_HEX_TAG | JSON_HEX_AMP | JSON_UNESCAPED_UNICODE
 		);
 
-		wp_add_inline_script(
-			self::SCRIPT_HANDLE,
-			sprintf( 'window.JetpackScriptData = %s;', $script_data ),
-			'before'
-		);
+		if ( is_admin() || did_action( 'enqueue_block_editor_assets' ) ) {
+			wp_enqueue_script( self::SCRIPT_HANDLE );
+			wp_add_inline_script( self::SCRIPT_HANDLE, "window.JetpackScriptData = {$script_json};", 'before' );
```

Comment:
Why was this changed from using a `sprintf` to add the JSON data to the output to adding it directly in the string?

Reply:
Good question, I'll answer in the next conversation below :)